automation/pxf_regress/main.go

```diff
@@ -0,0 +1,427 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+)
+
+// Options settable from command line
+var testDir string = "."
+var initFile string
+
+// gpdiff.pl options for comparing actual results file with expected
+//
+// Unlike postgres/pg_regress, PXF includes
+//
+//	-w ignore all white space
+//	-B ignore changes lines are all blank
+//
+// TODO: rather than match/sub DETAIL (GP5) for CONTEXT (see global_init_file), should we add "-I DETAIL:" and "-I CONTEXT:"
+// TODO: rather than having to add start_ignore/end_ignore, should we add "-I HINT:"
+var baseDiffOpts []string = []string{"-w", "-B", "-I", "NOTICE:", "-I", "GP_IGNORE", "-gpd_ignore_headers", "-U3"}
+
+// Wrap os.Stdout in bufio.Writer so that we can flush stdout when writing
+// status messages.
+var stdout *bufio.Writer
+var progName string
+
+// internal variables
+var gpdiffProg string
+
+// summary file of all diffs
+var regressionDiffsName string
+var regressionDiffs *os.File
+
+var regressionSummaryName string
+var regressionSummary *os.File
+
+var successCount int = 0
+var failCount int = 0
+
+func init() {
+	stdout = bufio.NewWriter(os.Stdout)
+}
+
+func main() {
+	pathName, err := os.Executable()
+	if err != nil {
+		fatalf(err.Error())
+	}
+	progName = pathName
+
+	if len(os.Args) < 2 {
+		fatalf("%s: missing required argument: test-directory", progName)
+	}
+
+	f, err := findOtherFile("gpdiff.pl", true)
+	if err != nil {
+		fatalf("%s: cannot find gpdiff.pl: %s", progName, err.Error())
+	}
+	gpdiffProg = f
+
+	f, err = findOtherFile("global_init_file", false)
```

Comment:
```suggestion
	f, err = findFile("global_init_file", false)
```
should we simplify the function name  now that "executability" requirement is a parameter ?